### PR TITLE
WT-2673 Stop automatically increasing memory page max

### PR DESCRIPTION
**Pull request by agorrod**

### CODE REVIEW

**Comment by agorrod** [comment]
@michaelcahill Would you mind reviewing this please?

File: src/btree/bt_handle.c [hunk 1]
```diff
@@ -689,21 +689,21 @@ __btree_page_sizes(WT_SESSION_IMPL *session)
...
+	btree->maxmempage = cval.val;
 	if (!F_ISSET(conn, WT_CONN_CACHE_POOL)) {
-		if ((cache_size = conn->cache_size) > 0)
+		if ((cache_size = conn->cache_size) > 0) {
+			btree->maxmempage =
+			    WT_MIN(btree->maxmempage, cache_size / 10);
+			/* Use a lower bound of a single disk leaf page */
 			btree->maxmempage =
-			    WT_MIN(btree->maxmempage, cache_size / 4);
+			    WT_MAX(btree->maxmempage, btree->maxleafpage);
```
Comment (michaelcahill) [inline src/btree/bt_handle.c]: @agorrod, I don't disagree with the change, but shouldn't we apply the `maxleafpage` lower bound outside this condition about cache size?

**Comment by michaelcahill** [comment]
Other than the ordering issue I mentioned, this change lgtm.

**Comment by agorrod** [comment]
Thanks @michaelcahill I've made the change you suggested. I'm going to go ahead and merge.

File: src/btree/bt_handle.c [hunk 2]
```diff
@@ -689,22 +689,20 @@ __btree_page_sizes(WT_SESSION_IMPL *session)
 		    "size (%" PRIu32 "B)", btree->allocsize);
 
 	/*
-	 * When a page is forced to split, we want at least 50 entries on its
-	 * parent.
-	 *
 	 * Don't let pages grow larger than a quarter of the cache, with too-
```
Comment (dmitri-shubin) [inline src/btree/bt_handle.c]: It seems to be one tenth now, not quarter.

Comment (agorrod) [inline src/btree/bt_handle.c]: Thanks - fixed.

**Pull request merged**